### PR TITLE
Don't emit the triplet warning with x-set-installed no args.

### PR DESCRIPTION
**Pull request by BillyONeal**
`x-set-installed` with no arguments should not emit this warning, but it was. I think it was thinking that was equivalent to install with no arguments which does need to warn.

Before and after:
bug demonstration

Also added e2e tests.

### CODE REVIEW

**Review by ras0219-msft — COMMENTED** [review]
Product changes LGTM

File: include/vcpkg-test/util.h [hunk 1]
```diff
@@ -143,16 +143,21 @@ namespace vcpkg::Test
         PackageSpec emplace(vcpkg::SourceControlFileAndLocation&& scfl);
     };
 
-    inline std::vector<FullPackageSpec> parse_test_fspecs(StringView sv, Triplet t = X86_WINDOWS)
+    inline std::vector<FullPackageSpec> parse_test_fspecs(StringView sv, bool expect_default_used)
```
[the remark this answers is not in the document]
Comment (BillyONeal) [reply include/vcpkg-test/util.h]: Done!

File: azure-pipelines/end-to-end-tests-dir/cli.ps1 [hunk 2]
```diff
@@ -9,3 +9,136 @@ Throw-IfNotFailed
 
 Run-Vcpkg -TestArgs ($commonArgs + @("install", "vcpkg-hello-world-1", "--fast")) # --fast is not a switch
 Throw-IfNotFailed
+
+if ($IsWindows) {
+    $warningText = 'Starting with the September 2023 release'
+
+    # build-external not tested
+    # ci not tested
+    # export not tested
```
Comment (dan-shaw) [inline azure-pipelines/end-to-end-tests-dir/cli.ps1:18]: Any way we can test if `VCPKG_DEFAULT_TRIPLET` environmental variable is set? This should not emit a warning message.

**Review by dan-shaw — APPROVED** [review]

File: azure-pipelines/end-to-end-tests-dir/cli.ps1 (hunk 2 above)
Comment (BillyONeal) [reply azure-pipelines/end-to-end-tests-dir/cli.ps1:18]: Done in 7166155f4b9509e3d1530a943295c5ebd33f4a0e !

**Pull request merged**